femwell/maxwell/waveguide.py

```diff
@@ -281,9 +282,128 @@ def plot(self, field, plot_vectors=False, colorbar=True, direction="y", title="E
+
+        if part == "real":
+            conv_func = np.real
+        elif part == "imag":
+            conv_func = np.imag
+        elif part == "abs":
+            conv_func = np.abs
+        else:
+            raise ValueError("A valid part is 'real', 'imag' or 'abs'.")
+
+        if ax is None:
+            fig = plt.figure()
+            ax = fig.add_subplot(111)
+
+        if field == "E":
+            mfield = self.E
+        elif field == "H":
+            mfield = self.H
+        else:
+            raise ValueError("A valid field is 'E' or 'H'.")
+
+        (mfield_t, mfield_t_basis), (mfield_n, mfield_n_basis) = self.basis.split(mfield)
+
+        if component == "x" or component == "y":
+            plot_basis = mfield_t_basis.with_element(ElementVector(ElementDG(ElementTriP1())))
+            mfield_xy = plot_basis.project(mfield_t_basis.interpolate(conv_func(mfield_t)))
+            (mfield_x, mfield_x_basis), (mfield_y, mfield_y_basis) = plot_basis.split(mfield_xy)
+            if component == "x":
+                mfield_x_basis.plot(mfield_x, ax=ax, shading="gouraud")
+            else:
+                mfield_y_basis.plot(mfield_y, ax=ax, shading="gouraud")
+        elif component == "t":
+            plot_basis = mfield_t_basis
+            mfield_t_basis.plot(conv_func(mfield_t), ax=ax, shading="gouraud")
+        elif component == "z" or component == "n":
+            plot_basis = mfield_n_basis
+            mfield_z, mfield_z_basis = mfield_n, mfield_n_basis
+            mfield_z_basis.plot(conv_func(mfield_z), ax=ax, shading="gouraud")
+        else:
+            raise ValueError("A valid component is 'x', 'y', 'z', 'n' or 't'.")
+
+        if boundaries:
+            for subdomain in plot_basis.mesh.subdomains.keys() - {"gmsh:bounding_entities"}:
+                plot_basis.mesh.restrict(subdomain).draw(ax=ax, boundaries_only=True, color="k")
+        if colorbar:
+            divider = make_axes_locatable(ax)
+            cax = divider.append_axes("right", size="5%", pad=0.05)
+            plt.colorbar(ax.collections[-1], cax=cax)
+
+        ax.set_title(f"{field}{component} ({part}. part)")
+
+        return ax
+
+    # def show(
+    #     self,
+    #     field: Literal["E", "H"] ,
+    #     part: Literal["real", "imag", "abs"] = "real",
+    #     boundaries: bool = True,
+    #     colorbar: bool = False,
+    # ):
+    #     fig, axs = plt.subplots(1, 3, subplot_kw=dict(aspect=1))
+
+    #     for id_ax, comp in enumerate("xyz"):
+    #         self.plot_component(field, comp, part, boundaries, colorbar, axs[id_ax])
+    #     plt.tight_layout()
+    #     plt.show()
+
+    def show(
+        self,
+        field: Literal["E", "H", "I"] | NDArray,
+        part: Literal["real", "imag", "abs"] = "real",
+        plot_vectors: bool = False,
+        boundaries: bool = True,
+        colorbar: bool = False,
+        direction: Literal["x", "y"] = "x",
+        title: str = "E",
```

Comment:
```suggestion
        title: Optional[str] = None,
```
That way we could use field (if it's a Literal) as the title in case the user doesn't define anything?

Reply:
just as a small extra: when you pass it on to the new function, could you add there for the title 
```python
title = field if title is None else title
```

Reply:
I am not sure about this one. If the user does not want to add a title, should we really enforce one?